concordium-sdk/src/main/java/com/concordium/sdk/responses/blocksummary/updates/queues/UpdateQueues.java

```diff
@@ -0,0 +1,159 @@
+import com.concordium.sdk.responses.blocksummary.updates.chainparameters.rewards.TransactionFeeDistribution;
+import com.concordium.sdk.responses.blocksummary.updates.keys.Level1KeysUpdates;
+import com.concordium.sdk.responses.blocksummary.updates.keys.Level2KeysUpdates;
+import com.concordium.sdk.responses.blocksummary.updates.keys.RootKeysUpdates;
+import com.concordium.sdk.transactions.CCDAmount;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import lombok.Getter;
+import lombok.ToString;
+
+/**
+ * Enqueued updates.
+ */
+// UpdateQueues must match
+// https://github.com/Concordium/concordium-base/blob/23c2f4f2fb524e60548f052d0422b567ec977822/haskell-src/Concordium/Types/UpdateQueues.hs#L119
+//
+// Note. bakerStakeThreshold for versions < 4 and 'poolParameters' for versions > 3
+//
+@Getter
+@ToString
+public final class UpdateQueues<PoolParameters> {
+
+    /**
+     * Root keys updates enqueued.
+     */
+    private final EnqueuedUpdate<RootKeysUpdates> rootKeysUpdates;
+
+    /**
+     * Level1 keys updates enqueued.
+     */
+    private final EnqueuedUpdate<Level1KeysUpdates> level1KeysUpdates;
+
+    /**
+     * Level2 keys updates enqueued.
+     */
+    private final EnqueuedUpdate<Level2KeysUpdates> level2KeysUpdates;
+
+    /**
+     * Protocol updates enqueued.
+     */
+    private final EnqueuedUpdate<ProtocolUpdate> protocolUpdates;
+
+    /**
+     * Election difficulty updates enqueued.
+     */
+    private final EnqueuedUpdate<Double> electionDifficultyUpdates;
+
+    /**
+     * Euro:energy exchange rate updates enqueued.
+     */
+    private final EnqueuedUpdate<Fraction> euroPerEnergyUpdates;
+
+    /**
+     * GTU:euro exchange rate updates enqueued.
+     */
+    private final EnqueuedUpdate<Fraction> microGTUPerEuroUpdates;
+
+    /**
+     * Foundation updates enqueued.
+     */
+    private final EnqueuedUpdate<Integer> foundationAccountUpdates;
+
+    /**
+     * Mint distribution updates enqueued.
+     */
+    private final EnqueuedUpdate<MintDistribution> mintDistribution;
+
+    /**
+     * Transaction fee distribution updates enqueued.
+     */
+    private final EnqueuedUpdate<TransactionFeeDistribution> transactionFeeDistribution;
+
+    /**
+     * Rewards updates enqueued.
+     */
+    private final EnqueuedUpdate<GasRewards> gasRewards;
+
+    /**
+     * Baker stake threshold.
+     */
+    private final EnqueuedUpdate<CCDAmount> bakerStakeThreshold;
+
+    /**
+     * Pool parameter updates enqueued.
+     */
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE, defaultImpl = PoolParametersV0.class)
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = PoolParametersV1.class)
+    })
+    private final EnqueuedUpdate<PoolParameters> poolParameters;
```

Comment:
Is this meant to cover both chain parameter versions V0 and V1? If so, I think the bakerStakeThreshold above is not necessary.

Reply:
Or this is using exactly the pool parameters in V1?

Reply:
I tightened up the type a bit.